core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -6677,6 +6677,32 @@ public boolean isBangEqualAllowed() {
         .withConformance(lenient).fails("Column 'E' not found in any table");
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6939">
+   * [CALCITE-6939] Add support for Lateral Column Alias</a>. */
+  @Test void testAliasInSelect() {
+    final SqlConformance withSelectAlias = new SqlAbstractConformance() {
+      @Override public boolean isSelectAlias() {
+        return true;
+      }
+    };
+
+    // Standard conformance
+    sql("select 1 AS x, ^x^ as Y")
```

Comment:
1: Does this currently support SQL execution? Or is it only supporting validate?
2: If the table includes x and y columns, what would the output be if it is renamed as x?
Some cases I test in Spark:
```
create table table(x int,y int);
insert into table values(1, 2);
select 2 as x, x + 2  from table; // output=> 2       3
 select 2 as x1, x1 + 2; // Error in query: cannot resolve '`x1`' given input columns: []; line 1 pos **17;**
```

Reply:
I have added some SqlOperatorTests to show that this works end-to-end.

Reply:
I tested it again using 3.5.1 and it is supported. Previously, I used 3.1.2.